### PR TITLE
feat(gui-v2): duration options

### PR DESCRIPTION
**Pull request by wingkwong**
## Change Summary

ref: #2953 

- add duration options
- rename durationHelper to durationOptions to keep the naming consistent

## Change type

- [x] feat: (new feature for the user, not a new feature for build script)
- [ ] fix: (bug fix for the user, not a fix to a build script)
- [ ] docs: (changes to the documentation)
- [ ] style: (formatting, missing semi colons, etc; no production code change)
- [ ] refactor: (refactoring production code, eg. renaming a variable)
- [ ] test: (adding missing tests, refactoring tests; no production code change)
- [ ] chore: (updating grunt tasks etc; no production code change)

## Test/ Verification

image

### CODE REVIEW

File: packages/nc-gui-v2/components/cell/Duration.vue [hunk 1]
```diff
@@ -1,7 +1,7 @@
 <script setup lang="ts">
 import { computed, inject, ref } from '#imports'
 import { ColumnInj } from '~/context'
-import { convertDurationToSeconds, convertMS2Duration, durationOptions } from '~/utils/durationHelper'
+import { convertDurationToSeconds, convertMS2Duration, durationOptions } from '~/utils/durationUtils'
```
Comment (pranavxc) [inline packages/nc-gui-v2/components/cell/Duration.vue:4]: Can be replace with 

```

import { convertDurationToSeconds, convertMS2Duration, durationOptions } from '~/utils'
```

**Review by pranavxc — CHANGES_REQUESTED** [review]

**Review by pranavxc — APPROVED** [review]

File: packages/nc-gui-v2/components/cell/Duration.vue (hunk 1 above)
Comment (bcakmakoglu) [inline packages/nc-gui-v2/components/cell/Duration.vue:4]: Import can still be reduced to just `~/utils` 😄

File: packages/nc-gui-v2/components/smartsheet-column/DurationOptions.vue [hunk 2]
```diff
@@ -0,0 +1,31 @@
+<script setup lang="ts">
+import { useColumnCreateStoreOrThrow } from '#imports'
+import { durationOptions } from '@/utils'
+
+const { formState, validateInfos, setAdditionalValidations, sqlUi, onDataTypeChange, onAlter } = useColumnCreateStoreOrThrow()
+
+const durationOptionList = computed(() =>
```
Comment (bcakmakoglu) [inline packages/nc-gui-v2/components/smartsheet-column/DurationOptions.vue:7]: unnecessary computed value, can just be a regular `const`

File: packages/nc-gui-v2/components/smartsheet-column/DurationOptions.vue [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+  durationOptions.map((o: Record<string, any>) => ({
```
Comment (bcakmakoglu) [inline packages/nc-gui-v2/components/smartsheet-column/DurationOptions.vue:8]: Unnecessary typecast to `Record<string, any>`

File: packages/nc-gui-v2/components/smartsheet-column/DurationOptions.vue [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+
+<template>
+  <a-row>
+    <a-col :span="24">
+      <span class="prose-sm mt-2">A duration of time in minutes or seconds (e.g. 1:23).</span>
+    </a-col>
+    <a-col :span="24">
+      <a-form-item label="Duration Format">
+        <a-select v-model:value="formState.meta.duration" size="small" class="w-52">
+          <a-select-option v-for="(duration, i) in durationOptionList ?? []" :key="i" :value="duration.id">
```
Comment (bcakmakoglu) [inline packages/nc-gui-v2/components/smartsheet-column/DurationOptions.vue:24]: use `for ... of ...`

Comment (bcakmakoglu) [reply packages/nc-gui-v2/components/smartsheet-column/DurationOptions.vue:24]: ?? [] is unnecessary

**Review by bcakmakoglu — CHANGES_REQUESTED** [review]

**Review by bcakmakoglu — APPROVED** [review]

**Pull request merged**